enumerable.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop:disable Layout/LineLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable  Metrics/MethodLength
+# rubocop:disable  Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    arr = *self
+    n.times do |i|
+      arr[i] = yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    n.times do |index|
+      yield(self[index], index)
+    end
+    self
+  end
+
+  def my_select
+    arr = []
+    my_each do |elem|
+      arr << elem if yield(elem) == true
+    end
+    arr
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each do |elem|
+        return false if yield(elem) == false
+      end
+    elsif !pattern.nil?
+      my_each do |elem|
+        unless (pattern == elem) || (pattern.is_a?(Class) && elem.is_a?(pattern)) || (pattern.is_a?(Regexp) && pattern.match?(elem.to_s))
+          return false
+        end
+      end
+    end
+    true
+  end
```

Comment:
- [ ] If the block given returns false or nil. Then it should return false

example: `[nil, true, 99].all?                              #=> false`